src/BPE.cc

```diff
@@ -28,6 +33,29 @@ namespace onmt
 
     int i = 0;
 
+    std::getline(in, line);
+
+    std::vector<std::string> options;
+    std::string option;
+
+    size_t sep = line.find(';');
+    size_t bidx = 0;
+    while (sep != std::string::npos && sep + 1 < line.size())
+    {
+      options.push_back(line.substr(bidx, sep-bidx));
+      bidx = sep + 1;
+      sep = line.find(';', bidx);
+    }
+    options.push_back(line.substr(bidx));
+
+    if (options.size() == 6 && options[0] == "v3")
+    {
+      BPE::prefix = (options[1] == "true");
+      BPE::suffix = options[2] == "true";
+      BPE::case_insensitive = options[3] == "true";
+      BPE::begin_of_word = options[4];
+      BPE::end_of_word = options[5];
+    }
```

Comment:
If options are not present in the BPE model, the first line is actually discarded. You can use:

```
in.seekg(0)
```

to come back at the beginning of the file.